### PR TITLE
node: aptos watcher

### PR DESCRIPTION
**Pull request by kcsongor**

### CODE REVIEW

**Comment by kcsongor** [comment]
Current dependencies on/for this PR:
* dev.v2
  * **PR #1731** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/1731" target="_blank">Graphite</a> 
    * **PR #1732** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/1732" target="_blank">Graphite</a> 
      * **PR #1733** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/1733" target="_blank">Graphite</a> 
        * **PR #1734** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/1734" target="_blank">Graphite</a> 
          * **PR #1735** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/1735" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/1735?utm_source=stack-comment).

File: Dockerfile.const [hunk 1]
```diff
@@ -11,6 +11,7 @@ RUN if [ -e /certs/cert.pem ]; then cp /certs/cert.pem /etc/pki/tls/certs/ca-bun
 # fetch scripts/guardian-set-init.sh deps
 RUN dnf -y install jq
 RUN dnf -y install make
+RUN dnf -y install findutils
```
Comment (jumpsiegel) [inline Dockerfile.const:14]: Acting as the ghost of @SEJeff past, he would suggest we combine these three lines into a single install call to reduce the size of the Dockerfile images.. maybe for another PR when we finally speed up Dockerfile.const

**Review by jumpsiegel — APPROVED** [review]

File: sdk/vaa/structs.go [hunk 2]
```diff
@@ -120,6 +120,10 @@ func (c ChainID) String() string {
 		return "algorand"
 	case ChainIDNear:
 		return "near"
+	case ChainIDAptos:
```
Comment (bruce-riley) [inline sdk/vaa/structs.go:123]: These should be added to structs_tests.go as well.

**Comment by bruce-riley** [comment]
Needs to be added to adminnodes.go as well.

File: node/pkg/watchers/aptos/watcher.go [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return body, err
+}
+
+func (e *Watcher) observeData(logger *zap.Logger, data gjson.Result, native_seq uint64) {
+	em := data.Get("sender")
+	if !em.Exists() {
+		logger.Info("sender")
```
Comment (bruce-riley) [inline node/pkg/watchers/aptos/watcher.go:82]: Should this message be an error or more informative?

File: sdk/vaa/structs.go (hunk 2 above)
Comment (jumpsiegel) [reply sdk/vaa/structs.go:123]: addressed

File: node/pkg/watchers/aptos/watcher.go (hunk 3 above)
Comment (jumpsiegel) [reply node/pkg/watchers/aptos/watcher.go]: Fixed

File: node/cmd/guardiand/node.go [hunk 4]
```diff
@@ -639,6 +659,9 @@ func runNode(cmd *cobra.Command, args []string) {
 		if *arbitrumContract != "" && !*unsafeDevMode {
 			logger.Fatal("Please do not specify --arbitrumContract")
 		}
+		if *aptosRPC != "" && !*unsafeDevMode {
```
Comment (bruce-riley) [inline node/cmd/guardiand/node.go:662]: Sorry, I think I missed this the first time. Doesn't this mean you won't be able to enable it in mainnet?

Comment (jumpsiegel) [reply node/cmd/guardiand/node.go]: Fixed.. amusingly, @kcsongor and I looked at this and confused ourselves on if we had to change it

**Review by bruce-riley — APPROVED** [review]

**Review by jumpsiegel — APPROVED** [review]

**Review by bruce-riley — APPROVED** [review]

**Review by jumpsiegel — DISMISSED** [review]

**Review by bruce-riley — DISMISSED** [review]

**Review by bruce-riley — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]
Proxy approving as only a test fix change was made.

**Pull request merged**